extensions/azurePublishNew/src/node/index.ts

```diff
@@ -1,51 +1,91 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { IExtensionRegistration } from '@botframework-composer/types';
+import { IBotProject, IExtensionRegistration, ProcessStatus, PublishPlugin } from '@botframework-composer/types';
 
-// TODO: remove this code eventually, added it so that extension would successfully build
+import { availableResources, setUpProvisionService } from './availableResources';
+import { BackgroundProcessManager } from './backgroundProcessManager';
+import { GetResourcesResult, ProvisionConfig, ResourceDefinition, ResourceProvisionService } from './types';
 
-function initialize(registration: IExtensionRegistration) {
-  const plugin1 = {
+/**
+ * Creates the azure publishing plug-in for this extension.
+ * @returns The plug-in with properties and methods.
+ */
+const createAzurePublishPlugin = (): PublishPlugin<ProvisionConfig> => {
```

Comment:
I created a factory function here rather than a class.  The goal is that each method implementation should be short and straightforward if we have architected and implemented the resources and services correctly.